faker/providers/python/__init__.py

```diff
@@ -198,6 +188,26 @@ def pydecimal(self, left_digits=None, right_digits=None, positive=False,
         else:
             sign = '+' if positive else self.random_element(('+', '-'))
 
+        if sign == '+':
+            if max_value is not None:
+                left_number = str(self.random_int(max(min_value or 0, 0), max_value))
+            else:
+                min_left_digits = math.ceil(math.log10(max(min_value or 1, 1)))
+                left_digits = left_digits or self.random_int(min_left_digits, max_left_random_digits)
```

Comment:
I believe this has inadvertently caused pydecimal to not work as intended when `left_digits=0`.